### ISSUE TITLE
Enhancement for EventType freeze

### ISSUE THREAD

**Issue by muthusamymm** [maksim-paskal/aks-node-termination-handler#43]
Hi,
Currently when azure sends eventType FREEZE , aks node terminator drains all pods  and stops watching for new events.
The issue what we see is , azure does not take that worker node down , so no new worker node creates by VMscaleset. 
The worker remains in unscheduled state and it is charged .

I possible for FREEZE state alone , after drain watch for events again and when the new event comes related to unfreeze/normal , uncordon that worker node and keep watching for new events .
![eventTypeFreeze](https://github.com/maksim-paskal/aks-node-termination-handler/assets/40357302/c01ac065-4939-46bc-b10f-5967159d6712)

**Comment by maksim-paskal** [comment]
@muthusamymm thanks for opening this issue. This make sense, Freeze is actualy temporary state for Azure resource.
```
Freeze: The Virtual Machine is scheduled to pause for a few seconds. CPU and network connectivity may be suspended, but there's no impact on memory or open files.

Reboot: The Virtual Machine is scheduled for reboot (non-persistent memory is lost). This event is made available on a best effort basis
Redeploy: The Virtual Machine is scheduled to move to another node (ephemeral disks are lost). This event is delivered on a best effort basis.
Preempt: The Spot Virtual Machine is being deleted (ephemeral disks are lost).
Terminate: The virtual machine is scheduled to be deleted.
```

I try to impelement some changes to exclude this event from draining

**Comment by muthusamymm** [comment]
Thank you for the support.

**Comment by maksim-paskal** [comment]
@muthusamymm released new version `v1.0.4` - in new version:
1) now Freeze event will not trigger node draining
2) log this events in node "events"
3) add Prometheus metrics, you could view this events now in Prometheus

you need to [install](https://github.com/maksim-paskal/aks-node-termination-handler#installation) new chart to get this changes

**Comment by muthusamymm** [comment]
Thank you for the update, will rollout in our env.
